grype/matcher/ruby/matcher.go

```diff
@@ -20,18 +20,6 @@ func (m *Matcher) Type() match.MatcherType {
 	return match.RubyGemMatcher
 }
 
-func (m *Matcher) Match(store vulnerability.Provider, _ *distro.Distro, p pkg.Package) ([]match.Match, error) {
-	var matches = make([]match.Match, 0)
-	langMatches, err := common.FindMatchesByPackageLanguage(store, p.Language, p, m.Type())
-	if err != nil {
-		return nil, err
-	}
-	matches = append(matches, langMatches...)
-
-	cpeMatches, err := common.FindMatchesByPackageCPE(store, p, m.Type())
-	if err != nil {
-		return nil, err
-	}
-	matches = append(matches, cpeMatches...)
-	return matches, nil
+func (m *Matcher) Match(store vulnerability.Provider, d *distro.Distro, p pkg.Package) ([]match.Match, error) {
+	return search.ByCriteria(store, d, p, m.Type(), search.CommonCriteria...)
```

Comment:
Nice work condensing this. It also makes it easier to consume as a library going forward.